x/stakeibc/keeper/host_zone.go

```diff
@@ -95,22 +95,12 @@ func (k Keeper) GetAllHostZone(ctx sdk.Context) (list []types.HostZone) {
 	return
 }
 
-func (k Keeper) AddDelegationToValidator(ctx sdk.Context, hostZone types.HostZone, valAddr string, amt int64) (success bool) {
+func (k Keeper) AddDelegationToValidator(ctx sdk.Context, hostZone types.HostZone, valAddr string, amt uint64) (success bool) {
 	for _, val := range hostZone.GetValidators() {
 		k.Logger(ctx).Info(fmt.Sprintf("Validator %s %d %d", val.GetAddress(), val.GetDelegationAmt(), amt))
 		if val.GetAddress() == valAddr {
-			if amt >= 0 {
-				val.DelegationAmt = val.GetDelegationAmt() + cast.ToUint64(amt)
-				return true
-			} else {
-				absAmt := cast.ToUint64(-amt)
-				if absAmt > val.GetDelegationAmt() {
-					k.Logger(ctx).Error(fmt.Sprintf("Delegation amount %d is greater than validator %s delegation amount %d", absAmt, valAddr, val.GetDelegationAmt()))
-					return false
-				}
-				val.DelegationAmt = val.GetDelegationAmt() - absAmt
-				return true
-			}
+			val.DelegationAmt = val.GetDelegationAmt() + cast.ToUint64(amt)
```

Comment:
have we validated that amt is positive? if it's negative, it'll get zero'd 

https://go.dev/play/p/eeAM0EJLiOj

```
func main() {
	var negative int64
	negative = -1000
	fmt.Println("negative", negative)
	zero := cast.ToUint64(negative)
	fmt.Println("zero", zero)
}
```
outputs
```
negative -1000
zero 0

Program exited.
```

Reply:
Addressed per new cast fork and using cast.To____E